cmd/operator-sdk/add/api.go

```diff
@@ -84,19 +142,144 @@ Example:
 	}
 	apiCmd.Flags().BoolVar(&skipGeneration, "skip-generation", false,
 		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
+	apiCmd.Flags().StringVar(&helmChartRef, "helm-chart", "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path)")
+	apiCmd.Flags().StringVar(&helmChartVersion, "helm-chart-version", "",
+		"Specific version of the helm chart (default is latest versio)")
+	apiCmd.Flags().StringVar(&helmChartRepo, "helm-chart-repo", "",
+		"Chart repository URL for the requested helm chart")
 
 	return apiCmd
 }
 
 func apiRun(cmd *cobra.Command, args []string) error {
+
 	projutil.MustInProjectRoot()
+	log.Infof("Generating api version %s for kind %s.", apiVersion, kind)
+
+	operatorType := projutil.GetOperatorType()
```

Comment:
Now that we support different operator types, each with slightly different usages, we need to validate the flags.

For example, `--helm-chart` is only valid if the operator type is Helm.

See https://github.com/operator-framework/operator-sdk/blob/179d8d6c81f8a14ff386f7d71372859551f56162/cmd/operator-sdk/new/cmd.go#L439-L492